File: LAB1/main.cpp
#include <iostream>
#include <math.h>
using namespace std;
int main()
{
	double  a, b, c, s, area;
	while(1){
		cout << "Please enter 3 triangle sides:" << endl;
		cin >> a >> b >> c;
		if (a==-1 && b==-1 && c==-1)
			break;
		if (a<=0 || b<=0 || c<=0){
			cout << "The input is not legal" << endl;
			continue;
		}
		else if ((a+b)<c || (a+c)<b || (b+c)<a){
			cout << "The input is not legal" << endl;
			continue;
		}
		s = (a + b + c)/2;
		area = sqrt(s * (s-a) * (s-b) * (s-c));
		cout << "The area of this triangle is:" << area << endl;
	}
	return 0;
}

File: LAB1/main.py
import math

while True:
    print("Please enter 3 triangle sides:")
    a, b, c = map(float, input().split())
    
    if a==-1 and b==-1 and c==-1:
        break
    if a<=0 or b<=0 or c<=0:
        print("The input is not legal")
        continue
    if (a+b)<c or (a+c)<b or (b+c)<a:
        print("The input is not leagl")
        continue
    s = (a + b + c)/2
    area = math.sqrt(s * (s-a) * (s-b) * (s-c))
    print(f"The area of this tiangle is:{area}")

